### ISSUE TITLE
saving results after each epoch 

### ISSUE THREAD

**Issue by AmirAzz** [ultralytics/ultralytics#1567]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi 
Can I save the result of training after each epoch? I run my code in Collab because of using GPU and most of the time after several epochs the training terminated due to lack of GPU and I have to start training from the first! for example in the following code : !yolo task=detect mode=train model=yolov8m.pt data={dataset.location}/data.yaml epochs=10 imgsz=800 plots=True

**Comment by glenn-jocher** [comment]
@AmirAzz this is already the default. last.pt and best.pt are saved in your PROJECT/NAME directory after each epoch is complete.

**Comment by glenn-jocher** [comment]
@JustasBart hmm this might be a bit confusing, maybe we should update the default to `save_period=1`

**Comment by JustasBart** [comment]
@glenn-jocher I think that's the case @glenn-jocher... I'm happy to do a PR on this if it's easy enough to edit those docs?

**Comment by glenn-jocher** [comment]
@JustasBart yes please submit a PR. In retrospect I'm not sure why I set this to -1, strange design decision. But also needs to be changed in code. Just search repo-wide for save_period, it pops up a few times, twice in docs and once in code I think.

**Comment by fhasanm** [comment]
I cannot really find the save_period argument in my cfg, so when I try to train, it just says save_period is not a valid argument. What might be the issue? 
Here's my cfg for reference:


Printing '/usr/local/lib/python3.10/dist-packages/ultralytics/yolo/cfg/default.yaml'

agnostic_nms: false
augment: false
batch: 16
box: 7.5
boxes: true
cache: false
cfg: null
classes: null
close_mosaic: 10
cls: 0.5
conf: null
copy_paste: 0.0
cos_lr: false
data: null
degrees: 0.0
deterministic: true
device: null
dfl: 1.5
dnn: false
dropout: 0.0
dynamic: false
epochs: 100
exist_ok: false
fl_gamma: 0.0
fliplr: 0.5
flipud: 0.0
format: torchscript
half: false
hide_conf: false
hide_labels: false
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
image_weights: false
imgsz: 640
int8: false
iou: 0.7
keras: false
label_smoothing: 0.0
line_thickness: 3
lr0: 0.01
lrf: 0.01
mask_ratio: 4
max_det: 300
mixup: 0.0
mode: train
model: null
momentum: 0.937
mosaic: 1.0
name: null
nbs: 64
nms: false
opset: null
optimize: false
optimizer: SGD
overlap_mask: true
patience: 50
perspective: 0.0
plots: true
pretrained: false
project: null
rect: false
resume: false
retina_masks: false
save: true
save_conf: false
save_crop: false
save_hybrid: false
save_json: false
save_txt: false
scale: 0.5
seed: 0
shear: 0.0
show: false
simplify: false
single_cls: false
source: null
task: detect
translate: 0.1
v5loader: false
val: true
verbose: true
vid_stride: 1
visualize: false
warmup_bias_lr: 0.1
warmup_epochs: 3.0
warmup_momentum: 0.8
weight_decay: 0.0005
workers: 8
workspace: 4

**Comment by glenn-jocher** [comment]
@fhasanm hi there,

It appears that you are trying to use the `save_period` argument which is not listed in your config file (default.yaml). This argument dictates the frequency at which the model should save weights during training epochs. If it's not present in your config file, it will not be recognized as a valid argument during training.

To resolve this, you may need to manually add this argument into your config file and specify its value based on your requirement. For instance, if you want to save weights after every epoch, you should add `save_period: 1` to your config file.

Remember to always save and close the config file before you initiate another training session so your changes take effect. This should hopefully resolve your issue. 

Let us know how it goes or if you have further questions.
